src/main/java/org/folio/edge/sip2/repositories/CirculationRepository.java

```diff
@@ -62,7 +69,7 @@
         .createResource(checkinRequestData);
 
     return result
-        .otherwiseEmpty()
+        .otherwise(() -> null)
```

Comment:
Is CirculationRepository returning null here? Not sure what this statement does.

Reply:
Perhaps this should be refactored to be a bit more clear. What is happening here is that the `Future` has failed, but we don't want to send the failure back to the handler in `MainVerticle` since we need to return a message to indicate that the request failed. So, in the failure case, the `otherwise` will execute. Here, we are telling the otherwise to send a new `IRequest` to the `compose`, with the resource (`JsonObject`) set to `null`. That way, the code will execute as normal and anywhere `resource.getResource()` is called, `null` will return. It might be a bit too much lambda magic. 🤔

Reply:
I for one didn't understand this magic trick :)

Reply:
Is this case tested in the cannotCheckout() test?

Reply:
Yes. That test returns a failed `Future` from the mock provider and we assert the `CheckoutResponse` has the data needed to create a SIP2 message indicating checkout failure.